### PR TITLE
feat: Zarf GPU config, README and name fixes

### PR DESCRIPTION
**Pull request by justinthelaw**
- Collapses everything into 1 image for CPU and GPU support
- Zarf deployment allocated GPU based on GPU_ENABLED flag
- Fixes for README and Deps

### CODE REVIEW

**Comment by justinthelaw** [comment]
Note: still need to fix the `config.yaml` for the new workflow.

**Review by gphorvath — DISMISSED** [review]

File: .github/workflows/release-docker-build-push-amd64.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
         run: git fetch --unshallow
 
       - name: Set VERSION
-        run: echo "VERSION=$(git describe --abbrev=0 --tags)" >> $GITHUB_ENV
+        run: echo "VERSION=$(git describe --abbrev=0 --tags | sed -e 's/^v//')" >> $GITHUB_ENV
```
Comment (YrrepNoj) [inline .github/workflows/release-docker-build-push-amd64.yaml:19]: Same comment about using `$GITHUB_REF_NAME` please

File: Dockerfile [hunk 2]
```diff
@@ -1,30 +1,39 @@
...
-
 WORKDIR /leapfrogai
 
-COPY requirements.txt .
+# download model
+RUN python -m pip install -U huggingface_hub[cli,hf_transfer]
+ARG REPO_ID=TheBloke/SynthIA-7B-v2.0-GGUF
+ARG FILENAME=synthia-7b-v2.0.Q4_K_M.gguf
+ARG REVISION=3f65d882253d1f15a113dabf473a7c02a004d2b5
+COPY scripts/ scripts/
```
Comment (YrrepNoj) [inline Dockerfile:13]: only copy the `model_download.py` script

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,11 @@
-VERSION := $(shell git describe --abbrev=0 --tags 2> /dev/null )
+MODEL_URL ?= https://huggingface.co/TheBloke/SynthIA-7B-v2.0-GGUF/resolve/main/synthia-7b-v2.0.Q4_K_M.gguf
```
Comment (YrrepNoj) [inline Makefile:1]: This can be deleted

File: zarf.yaml [hunk 4]
```diff
@@ -16,4 +16,4 @@ components:
     required: true
     import:
       name: model
-      url: oci://ghcr.io/defenseunicorns/packages/leapfrogai/leapfrogai-model:0.4.1
\ No newline at end of file
+      url: oci://ghcr.io/defenseunicorns/packages/leapfrogai/leapfrogai-model:0.5.0-skeleton
```
Comment (YrrepNoj) [inline zarf.yaml:19]: Note to self to make sure we merge the `leapfrog-model-skeleton` PR first so we can make sure this artifact exists.

File: pyproject.toml [hunk 5]
```diff
@@ -15,10 +15,11 @@ dev = [
     "pytest",
     "black",
     "isort",
-    "nvidia-cublas-cu12",
```
Comment (YrrepNoj) [inline pyproject.toml:18]: Can we pin the `reuqires-python` above to 3.11.6

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**